include/sys/dmu_objset.h

```diff
@@ -124,11 +124,13 @@ struct objset {
 
 	/* Protected by os_lock */
 	kmutex_t os_lock;
-	list_t os_dirty_dnodes[TXG_SIZE];
-	list_t os_free_dnodes[TXG_SIZE];
```

Comment:
I'm happy to see `os_free_dnodes` go.  I've found it confusing in the past since care is taken to maintain them for no apparent purpose.